Remotion/Web/ClientScript/Scripts/TreeView.ts

```diff
@@ -155,13 +156,14 @@ class WebTreeView
           let hasChildren = activeTreeNode.getAttribute ('aria-expanded') === 'true';
           if (!hasChildren)
           {
-            let parent = activeTreeNode.parentElement;
+            let index = treeNodes.indexOf(activeTreeNode);
+            let parent = parentTreeNodeListItems[index].parentElement;
             while (parent && !parent.matches ('ul[role=tree]'))
             {
-              if (parent.tagName.toLowerCase() === 'li')
+              if (parent.matches('li:has(a[role=treeitem])'))
               {
-                let newTreeNode = parent;
-                WebTreeView.SetFocus (newTreeNode, activeTreeNode);
+                let nodeIndex = parentTreeNodeListItems.indexOf(parent);
+                WebTreeView.SetFocus (treeNodes[nodeIndex], activeTreeNode);
                 break;
               }
 
```

Comment:
I'm trying to think through the updated logic and I'm not sure if it couldn't be simplified:
1. take activeTreeNode and set as loop-variable (`parent`)
2. walk up the parent until you hit an `UL` with `role=tree`
3. when you find a `LI`, stop and select the treenode-Anchor inside.

matching `LI` with nested a with a role=treeitem seems overkill.
I also think we can keep the original stucture and simply change the way the `newTreeNode` is picked. Here, if you go with the array or do a fresh selection is simply a performance question and it could very well be that selecting the `A`-child is cheaper than the hashtable lookup. Or at least, won't be felt and we can keep it more straight forward (that's TBD)

Reply:
we can't take activeTreeNode and set as a parent, as this is still contained in a different LI which would be matched.

Matching with the nested a + role is probably overkill.